libraries/cloudharness-common/cloudharness/auth/keycloak/__init__.py

```diff
@@ -252,3 +252,49 @@ def current_user_has_client_role(self, client_name, role):
             self._get_keycloak_user_id(),
             client_name,
             role)
+
+    def user_add_update_attribute(self, user_id, attribute_name, attribute_value):
+        """
+        Adds or when exists updates the attribute to/of the User with the attribute value
+
+        param user_id: id of the user
+        param attribute_name: name of the attribute to add/update
+        param attribute_value: value of the attribute
+        :return: boolean True on success
+        """
+        admin_client = self.get_admin_client()
+        try:
+            user = self.get_user(user_id)
+            attributes = user.get('attributes', [])
+            attributes[attribute_name] = attribute_value
+            admin_client.update_user(
+                user_id,
+                {
+                    'attributes': attributes
+                })
+            return True
+        except:
+            return False
```

Comment:
I think that it would be better to wrap and rethrow the exception. Using the return value to state success is so old C-like!

Reply:
done